README.md

```diff
@@ -0,0 +1,152 @@
+
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+rules_detekt_version = "see-github-releases-page"
+rules_detekt_sha256 = "see-github-releases-page"
+
+http_archive(
+    name = "rules_detekt",
+    urls = ["https://github.com/buildfoundation/bazel_rules_detekt/archive/%s.zip" % rules_detekt_version],
+    type = "zip",
+    strip_prefix = "rules_detekt-%s" % rules_detekt_version,
+    sha256 = rules_detekt_sha256,
+)
+```
+
+### `BUILD` Configuration
+
+Once declared in `WORSKPACE`, `rules_detekt` can be `load`ed and applied in a `BUILD` file:
+
+```python
+load("@rules_detekt//detekt:detekt.bzl", "detekt")
+
+detekt(
+    name = "my_detekt",
+    srcs = glob(["src/main/kotlin/**/*.kt"]),
+)
+```
+
+### Executing
+
+Once set up, Detekt targets can be ran with `bazel build //mypackage:my_detekt`.
+Detekt targets will be cached if ran successfully.
+
+### Supported Attributes
+
+#### `srcs`
+
+- Starlark type: `label_list`
+- Mandatory: `True`
+
+A glob or an explicit list of `.kt` files for Detekt analysis.
+
+```python
+detekt(
+    srcs = glob(["src/main/kotlin/**/*.kt"]),
+)
+```
+
+#### `config`
+
+- Starlark type: `label`
+- Mandatory: `False`
+- Default value: [default Detekt configuration](https://github.com/arturbosch/detekt/blob/master/detekt-cli/src/main/resources/default-detekt-config.yml) (mind the Detekt version)  
+
+A path or target that represents custom Detekt configuration file.
+File must have `.yml` extension (expected by Detekt).
+See official Detekt documentation for details https://github.com/arturbosch/detekt/blob/master/detekt-cli/src/main/resources/default-detekt-config.yml
+
+```python
+detekt(
+    config = "my-detekt-config.yml",
+)
+```
+
+#### `parallel`
+
+- Starlark type: `bool`
+- Mandatory: `False`
+- Default value: `False`
+
+As per Detekt documentation https://arturbosch.github.io/detekt/cli.html: Enables/disables parallel compilation of source files. 
+Should only be used if the analyzing project has more than ~200 Kotlin files.
+
+```python
+detekt(
+    parallel = True,
+)
+```
+
+#### `xml_report`
+
+- Starlark type: `bool`
+- Mandatory: `False`
+- Default value: `False`
+
+Enables/disables XML report file generation. The report is generated with following pattern: `{target_name}_detekt_report.xml`
+FYI Detekt uses Checkstyle XML reporting format which makes it compatible with tools like SonarQube and so on. 
+Note that `rules_detekt` always generate txt report file: `{target_name}_detekt_report.txt`
+
+```python
+detekt(
+    xml_report = True,
+)
+```
+
+#### `html_report`
+
+- Starlark type: `bool`
+- Mandatory: `False`
+- Default value: `False`
+
+Enables/disables HTML report file generation. The report is generated with following pattern: `{target_name}_detekt_report.html` 
+Note that `rules_detekt` always generate txt report file: `{target_name}_detekt_report.txt`
+
+```python
+detekt(
+    html_report = True,
+)
+```
```

Comment:
All rules I saw so far (Apple, Swift, `jvm_external`) use tables and I suggest to steal this approach. Example:

Name | Starlark | Optional | Description
------|---------|----------|-----------
`html_report` | `bool` | Yes | Enables/disables HTML report file generation. The report is generated with following pattern: `{target_name}_detekt_report.html`. Note that `rules_detekt` always generate txt report file: `{target_name}_detekt_report.txt`

I guess GitHub doesn’t support snippets in table cells but we can make a dedicated Example section with all options applied. Other rules do that actually.

BTW, do you want to move options to a dedicated file at `docs`? Other rules do that.

Reply:
Not against making a table, but prob would prefer to keep it in README.md for discoverability since this ruleset is pretty simple and straighforward unlike other rules you referenced as an example

Feel free to submit this change!